cooltools/cli/compute_pileup.py

```diff
@@ -0,0 +1,260 @@
+    type=int,
+    default=None,
+    show_default=True,
+)
+@click.option(
+    "--aggregate",
+    help="Function for calculating aggregate signal. ",
+    default="none",
+    type=click.Choice(
+        ["none", "mean", "median", "std", "min", "max"], case_sensitive=False
+    ),
+    show_default=True,
+)
+@click.option("-f", "--force", help="Forced computation", is_flag=True, default=False)
+@click.option(
+    "-v", "--verbose", help="Enable verbose output", is_flag=True, default=False
+)
+def compute_pileup(
+    cool_path,
+    features,
+    view,
+    expected,
+    flank,
+    features_format,
+    weight_name,
+    out,
+    out_format,
+    store_snips,
+    nproc,
+    ignore_diags,
+    aggregate,
+    force,
+    verbose,
+):
+    """
+    Perform retrieval of the snippets from .cool file.
+
+    COOL_PATH : The paths to a .cool file with a balanced Hi-C map. Use the
+    '::' syntax to specify a group path in a multicooler file.
+
+    FEATURES_PATH : the path to a BED or BEDPE-like file that contains features for snipping windows.
+    If BED, then the features are on-diagonal. If BEDPE, then the features
+    can be off-diagonal (but not in trans or between different regions in the view).
+
+    """
+
+    clr = cooler.Cooler(cool_path)
+
+    #### Read the features:
+    buf, names = sniff_for_header(features)
+    if features_format.lower() == "bedpe":
+        default_cols = [0, 1, 2, 3, 4, 5]
+        bedpe_cols = ["chrom1", "start1", "end1", "chrom2", "start2", "end2"]
+        dtypes = {
+            "chrom1": str,
+            "start1": np.int64,
+            "end1": np.int64,
+            "chrom2": str,
+            "start2": np.int64,
+            "end2": np.int64,
+        }
+        if names is None:
+            kwargs = dict(
+                header=None,
+                usecols=default_cols,
+                dtype=dtypes,
+                names=bedpe_cols,
+            )
+        else:
+            kwargs = dict(header="infer", usecols=bedpe_cols)
+    elif features_format.lower() == "bed":
+        default_cols = [0, 1, 2]
+        bed_cols = ["chrom", "start", "end"]
+        dtypes = {"chrom": str, "start": np.int64, "end": np.int64}
+        if names is None:
+            kwargs = dict(
+                header=None,
+                names=bed_cols,
+            )
+        else:
+            kwargs = dict(header="infer", usecols=bed_cols)
+    else:
+        raise ValueError(
+            "Automatic detection of features format is not implemented yet. "
+            "Please provide BED or BEDPE as --features-format"
+        )
+
+    features_df = pd.read_table(
+        buf, comment="#", usecols=default_cols, dtype=dtypes, verbose=verbose, **kwargs
+    )
+
+    ###### Define view for cis compartment-calling
+    # use input "view" BED file or all chromosomes mentioned in "track":
+    if view is None:
+        # Generate viewframe from clr.chromsizes:
+        view_df = bioframe.make_viewframe(
```

Comment:
you can do `make_cooler_view(clr)` for that

Reply:
or maybe we could keep it as is - and I'll modify it using new functions from .lib.common afterwards

Reply:
I can update, thanks!